cms/grading/tasktypes/Communication.py

```diff
@@ -179,7 +179,7 @@ def evaluate(self, job, file_cacher):
             num_processes = 1
         else:
             num_processes = self.parameters[0]
-        indices = range(num_processes)
+        indices = list(range(num_processes))
```

Comment:
I don't think so, it is iterated upon several times in the following, if it were an iterator it would be "exhausted" after the first.

Reply:
Uhm...

```In [15]: a = xrange(3)

In [16]: print [x for x in a]
[0, 1, 2]

In [17]: print [x for x in a]
[0, 1, 2]

In [18]: print [x for x in a]
[0, 1, 2]

In [19]: print [x for x in a]
[0, 1, 2]
```

Works also in py3k with range - but I guess it's ok if you want to keep it, since is supersmall

Reply:
Wow, didn't think that would work, I will fix this, thanks!